catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -43,99 +42,94 @@
 
 public class ImagingChipActionProvider implements MultiActionProvider {
 
-  public static final String TITLE = "Chip Image";
+  static final String TITLE = "Chip Image";
 
-  public static final String DESCRIPTION =
+  static final String DESCRIPTION =
       "Opens a new window to enter the boundaries of an image chip for a Metacard.";
 
-  public static final String PATH = "/chipping/chipping.html";
+  private static final String PATH = "/chipping/chipping.html";
 
-  public static final String ID = "catalog.data.metacard.image.chipping";
+  static final String ID = "catalog.data.metacard.image.chipping";
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ImagingChipActionProvider.class);
 
-  private static final String NITF_IMAGE_METACARD_TYPE = "isr.image";
+  static final String NITF_IMAGE_METACARD_TYPE = "isr.image";
 
   private static final String ORIGINAL_QUALIFIER = "original";
 
   private static final String QUALIFIER_KEY = "qualifier";
 
-  private GeometryFactory geometryFactory = new GeometryFactory();
+  private static final Pattern ALLIANCE_DOWNLOAD_RESOURCE_PATH_PATTERN =
+      Pattern.compile("/.*/catalog/sources/.*/.*");
+
+  private final GeometryFactory geometryFactory = new GeometryFactory();
 
   @Override
   public <T> List<Action> getActions(T input) {
-    if (input == null) {
-      LOGGER.debug("Metacard can not be null.");
-      return new ArrayList<>();
-    }
+    final Metacard metacard = (Metacard) input;
```

Comment:
❓ Do we not want to double check the type of input before casting it? Even if to throw a little nicer exception than ClassCastException?

Reply:
`canHandle` is always called before `getActions`, and `canHandle` checks that the `input` is a `Metacard`. I made https://github.com/codice/ddf/pull/2472 to clarify that in the javadoc.

Reply:
Ah 👍